### ISSUE TITLE
After long time BS misbehaved

### ISSUE THREAD

**Issue by jilavsky** [APS-USAXS/ipython-usaxs#522]
10/16/2021 - tried using BS again, after long time not being used. Prior users had used spec (Diesel spray) so it was not used for ~5 days. Following issues happened:
1. Collecting data - SAXS worked fine, WAXS failed to collect data
2. Quit BS, command line left unresponsive
3. Created new terminal, reestarted BS, all works fine

no need to do anything with WAXS ioc, also, it was working just before this over night from spec. It is unlikely this is WAXS issue. 
 
Buffer from before failure:

```
I Sat-08:26:48 - sample image file: /share1/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_saxs/AgBehenatLaB6_0110.jpg
D Sat-08:26:48 - areaDetectorAcquire(): OrderedDict([('enable', 1), ('auto_increment', 'Yes'), ('array_counter', 0), ('auto_save', 'Yes'), ('num_capture', 0), ('file_template', '%s%s_%4.4d.hdf'), ('file_write_mode', 'Single'), ('blocking_callbacks', 'No'), ('parent.cam.array_callbacks', 1)])

Transient Scan ID: 6     Time: 2021-10-16 08:26:48
Persistent Unique Scan ID: 'c2f2c742-e304-442a-9b06-c50f60508062'
                                                                                                                                                                              
I Sat-08:26:53 - Finished SAXS/WAXS data collection in 5.639190196990967 seconds.
I Sat-08:26:53 - I0 value: 4253.0
I Sat-08:26:53 - memory report: pmem(rss=4567179264, vms=9035726848, shared=54677504, text=2506752, lib=0, data=8403255296, dirty=0)
I Sat-08:26:53 - file line 26:       waxsExp        15.5      49      1      "AgBehenatLaB6"
D Sat-08:26:54 - waxsx start collection =0.0
D Sat-08:26:54 - waxsx after before plan =0.0
I Sat-08:26:54 - Found UsaxsSaxsMode = SAXS in beam
I Sat-08:26:54 - Moving to proper WAXS mode
                                                                                                                                                                              
I Sat-08:27:01 - Removed SAXS from beam position
I Sat-08:27:01 - Moving USAXS out of beam
I Sat-08:27:01 - Removed USAXS from beam position
                                                                                                                                                                              
                                                                                                                                                                              
                                                                                                                                                                              
I Sat-08:27:11 - WAXS is in position
D Sat-08:27:11 - guard slits horizontal difference = 0
D Sat-08:27:11 - guard slits vertical difference = 0
D Sat-08:27:11 - USAXS slits horizontal difference = 7.77156e-15
D Sat-08:27:11 - USAXS slits vertical difference = 7.35e-05
I Sat-08:27:11 - Prepared for WAXS mode
D Sat-08:27:12 - waxsx after mode_WAXS =261.0
I Sat-08:27:12 - Creating technique directory: /share1/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs
I Sat-08:27:12 - Area Detector HDF5 file: /share1/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs/AgBehenatLaB6_0110.hdf
                                                                                                                                                                              
D Sat-08:27:13 - gain=10000000.0  rate: 86920.0  max: 950000  converged=[True, True]
                                                                                                                                                                              
D Sat-08:27:13 - gain=10000000.0  rate: 87000.0  max: 950000  converged=[True, True]
                                                                                                                                                                              
D Sat-08:27:13 - gain=10000000.0  rate: 87240.0  max: 950000  converged=[True, True]
D Sat-08:27:13 - gain=100000.0  rate: 270560.0  max: 950000  converged=[True, True, True, True]
                                                                                                                                                                              
                                                                                                                                                                              
I Sat-08:27:15 - sample image file: /share1/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs/AgBehenatLaB6_0110.jpg
D Sat-08:27:15 - areaDetectorAcquire(): OrderedDict([('enable', 1), ('auto_increment', 'Yes'), ('array_counter', 0), ('auto_save', 'Yes'), ('num_capture', 0), ('file_template', '%s%s_%4.4d.hdf'), ('file_write_mode', 'Single'), ('blocking_callbacks', 'No'), ('parent.cam.array_callbacks', 1)])
I Sat-08:27:20 - WAXS: 15.0s remaining
I Sat-08:27:25 - WAXS: 10.0s remaining
E Sat-08:27:25 - Exception TimeoutError during attempt 1 of 1 of command '('waxsExp', ['15.5', '49', '1', 'AgBehenatLaB6'], 26, '      waxsExp        15.5      49      1      "AgBehenatLaB6"')''
subject: TimeoutError during attempt 1 of 1 of command '('waxsExp', ['15.5', '49', '1', 'AgBehenatLaB6'], 26, '      waxsExp        15.5      49      1      "AgBehenatLaB6"')''

date: 2021-10-16 08:27:25.978234
command file: /share1/USAXS_data/2021-10/usaxs.mac
line number: 26
command: ('waxsExp', ['15.5', '49', '1', 'AgBehenatLaB6'], 26, '      waxsExp        15.5      49      1      "AgBehenatLaB6"')
raw command:       waxsExp        15.5      49      1      "AgBehenatLaB6"
attempt: 1 of 1
exception: Attempted to set EpicsPathSignal(read_pv='usaxs_pilatus2:HDF1:FilePath_RBV', name='waxs_det_hdf1_file_path', parent='waxs_det_hdf1', value='/mnt/usaxscontrol/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs/', timestamp=1634390835.8525255, auto_monitor=True, string=True, write_pv='usaxs_pilatus2:HDF1:FilePath', limits=False, put_complete=False, path_semantics='posix') to value '/mnt/usaxscontrol/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs/' and timed out after 10 seconds. Current value is '/mnt/usaxscontrol/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs'.
Stopping further processing of this command list.

Out[64]: 27:26 - memory report: pmem(rss=4567134208, vms=9035726848, shared=54677504, text=2506752, lib=0, data=8403255296, dirty=0)
('168ded6f-d344-455b-94fa-39ec94fd6bcc',
 'ba250091-81b6-46b5-9f8a-64ab523b1dec',
 '9ee95bb9-0426-43d4-963e-500b2390a7ae',
 'e2faf750-d3f1-4c87-abf4-952d7aad125e',
 'c2f2c742-e304-442a-9b06-c50f60508062')

In [65]: I Sat-08:27:30 - WAXS: 5.0s remaining
In [65]: 

In [65]: exit
usaxs@usaxscontrol .../USAXS_data/2021-10 $
```

after restart all is OK
```
In [2]: RE(run_command_file("usaxs.mac"))
I Sat-08:34:35 - Command file: usaxs.mac
====== ======================= ==========================
line # action                  parameters                
====== ======================= ==========================
1      CURRENT_EXPERIMENT_NAME NMRTubesSet11             
23     saxsExp                 15.5, 49, 1, AgBehenatLaB6
26     waxsExp                 15.5, 49, 1, AgBehenatLaB6
====== ======================= ==========================

I Sat-08:34:35 - memory report: pmem(rss=280322048, vms=3245268992, shared=46637056, text=2506752, lib=0, data=2718339072, dirty=0)
I Sat-08:34:36 - Command file verified
I Sat-08:34:36 - Measuring background for: PD_USAXS
                                                                                                                                                                              
I Sat-08:34:37 - PD_USAXS range=4 gain=1e12 bkg=18124.6 +/- 929.0544870996533
I Sat-08:34:37 - I0_USAXS range=4 gain=1e9 bkg=314.6 +/- 0.48989794855663565
I Sat-08:34:37 - I00_USAXS range=4 gain=1e9 bkg=512.8 +/- 0.4
    
```

NOTE memory issues??? see rss memory which after is 20x higher than after restart. Problem??? 

```
Out[64]: 27:26 - memory report: pmem(rss=4567134208, vms=9035726848, shared=54677504, text=2506752, lib=0, data=8403255296, dirty=0)
I Sat-08:34:35 - memory report:  pmem(rss=  280322048, vms=3245268992, shared=46637056, text=2506752, lib=0, data=2718339072, dirty=0)

```

**Comment by prjemian** [comment]
**we**?  I did not get that message.

**Comment by jilavsky** [comment]
You should, e-mail address is me, Ivan and you. Came from usaxs at 8:27am.

**Comment by prjemian** [comment]
The emailed error (after editing all the extra info):

```
Attempted to set usaxs_pilatus2:HDF1:FilePath_RBV (name='waxs_det_hdf1_file_path'
to value '/mnt/usaxscontrol/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs/'
timed out after 10 seconds
Current value is '/mnt/usaxscontrol/USAXS_data/2021-10/10_16_Pyush2/10_16_Pyush2_waxs'
```

note the difference in the trailing `/`

Why is that missing?

**Comment by jilavsky** [comment]
no idea... I will start new issue for what I see now, I think this is related to pausing for beam in bad moment.

**Comment by prjemian** [comment]
Yeah, bad timing for a beam time.  Recent rash of RF trips.

**Comment by jilavsky** [comment]
another beam dump

**Comment by jilavsky** [comment]
Observation:
1. I exited BS and restarted. Same problem happened again, same error.
2. I exited BS and I killed terminal (command lien it run in). I c reated NEW terminal session and started BS. Now all is Correct.

something is hanging after we exit BS and is having impact on new BS session after restart. 
this happened only after 2 days of using BS.

**Comment by prjemian** [comment]
That observation is valuable, thanks.
